### PR TITLE
make sure all code patch steps are validated

### LINKED ISSUE #28: Fail build (/throw errors) when code patching fails
In various places we do some code patching that is completely necessary for the worker to run, for example:

https://github.com/opennextjs/opennextjs-cloudflare/blob/42320e78e4e8f1abafd2336c3a384fedc2f8a394/packages/cloudflare/src/build/patches/to-investigate/patch-find-dir.ts#L12-L22

But we do not check if such patches have been completed successfully (e.g. in the example above we never check wether `"function findDir(dir, name) {"` is actually found and replaced in the code or not)

In order not to complete the build process in a seemingly successful way just to end up with a non-fully (or not-at-all) functional worker we should throw errors and fail the build process in case some of these patches aren't getting successfully applied.

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
resolves #28 

___

This PR makes sure that all our code patches are applied (that they do change the worker's code)

As part of this I did a bit of refactoring to remove every single function's log and validation and consolidate then
in a single function `patchCodeWithValidations`

___

The result/logs in case everything works as expected:
Screenshot 2024-12-18 at 10 25 37

The result/logs in case a step fails:
Screenshot 2024-12-18 at 10 31 17

### CODE REVIEW

File: packages/cloudflare/src/cli/build/bundle-server.ts [hunk 1]
```diff
@@ -139,6 +128,7 @@ globalThis.__dangerous_ON_edge_converter_returns_request = true;
     );
   }
 
+  console.log();
```
Comment (vicb) [inline packages/cloudflare/src/cli/build/bundle-server.ts:131]: It's this intended?

File: packages/cloudflare/src/cli/build/bundle-server.ts [hunk 2]
```diff
@@ -192,3 +192,33 @@ function createFixRequiresESBuildPlugin(config: Config): Plugin {
...
+ * @param code the code to apply the patches to
+ * @param patches array of functions that take a string (pre-patch code) and return a string (post-patch code)
+ * @returns the patched code
+ */
+async function patchCodeWithValidations(
+  code: string,
+  patches: [string, (code: string) => string | Promise<string>][]
+): Promise<string> {
+  console.log(`Applying code patches:`);
+  let prePatchCode = code;
```
Comment (vicb) [inline packages/cloudflare/src/cli/build/bundle-server.ts:209]: Nit: you can remove this line and add a let l215

**Review by vicb — APPROVED** [review]
Thanks!

File: packages/cloudflare/src/cli/build/bundle-server.ts (hunk 1 above)
Comment (dario-piotrowicz) [reply packages/cloudflare/src/cli/build/bundle-server.ts:131]: Yes, for this empty line:
Screenshot 2024-12-18 at 11 36 06

probably a `\n` in the other console.log would be better/clearer?

Comment (vicb) [reply packages/cloudflare/src/cli/build/bundle-server.ts:131]: Moving this log to l222 inside `patchCodeWithValidations` as "Applied N patches" might be a bit clearer, up to you.

Comment (dario-piotrowicz) [reply packages/cloudflare/src/cli/build/bundle-server.ts:131]: sounds good, I don't have a strong preference here 🙂

**Pull request merged**